### PR TITLE
Removes the need to pass DOB and Type to update Legal Entity

### LINKED ISSUE #308: Can't update legal entity without DOB
It default to 0 and Stripe API returns a validation error.
It also sends empty string for the type, which is unexpected.

```go
params := &stripe.AccountParams{
  LegalEntity: &stripe.LegalEntity{
    PersonalID: "000-00-0000",
  },
}

account.Update(req.AccountID, params)
```

```
&{values:[
  {Key:legal_entity[type] Value:} 
  {Key:legal_entity[dob][day] Value:0} 
  {Key:legal_entity[dob][month] Value:0} 
  {Key:legal_entity[dob][year] Value:0} 
  {Key:legal_entity[personal_id_number] Value:000-00-0000} 
]}
```

```json
{"field_name":"legal_entity[dob][day]", "code":"", "messages":["Not a valid day"]}
```

### PR DESCRIPTION
**Pull request by guillaumehenriot**
It fixes #308.

Prior to that we needed to pass Type and the full DOB to update Legal Entity. These parameters are now checked to see if empty like everything else.

r? @brandur

### CODE REVIEW

File: account/client_test.go [hunk 1]
```diff
@@ -238,6 +238,45 @@ func TestAccountUpdate(t *testing.T) {
...
+		},
+	}
+
+	acct, err = Update(acct.ID, params)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if acct.LegalEntity.Address.Line1 != params.LegalEntity.Address.Line1 {
+		t.Error("The account address line1 %s does not match the params address line1: %s", acct.LegalEntity.Address.Line1, params.LegalEntity.Address.Line1)
```
Comment (brandur) [inline account/client_test.go]: I think you're looking for `Errorf` here.

Comment (guillaumehenriot) [inline account/client_test.go]: Oops, you're right. I'll change that.

**Comment by brandur** [comment]
LGTM!

**Comment by brandur** [comment]
Released as 17.4.1.

**Pull request merged**